apps/backend/src/authn/gitlab.strategy.ts

```diff
@@ -0,0 +1,35 @@
+import {Injectable} from '@nestjs/common';
+import {PassportStrategy} from '@nestjs/passport';
+import {Strategy} from 'passport-gitlab2';
+import {ConfigService} from '../config/config.service';
+import {AuthnService} from './authn.service';
+
+@Injectable()
+export class GitlabStrategy extends PassportStrategy(Strategy, 'gitlab') {
+  constructor(
+    private readonly authnService: AuthnService,
+    private readonly configService: ConfigService
+  ) {
+    super({
+      clientID: configService.get('GITLAB_CLIENTID') || 'disabled',
+      clientSecret: configService.get('GITLAB_SECRET') || 'disabled',
+      callbackURL:
+        `${configService.get('EXTERNAL_URL')}/authn/gitlab/callback` ||
+        'disabled'
+    });
+  }
+
+  async validate(
+    accessToken: string,
+    refreshToken: string,
+    profile: {
+      [key: string]: any;
```

Comment:
Instead of `any` here, can you just add the fields you expect to receive?

Reply:
While I can't explicitly type all of the fields, I'll type the fields in use and leave the rest as any.